src/TS/services/runInfoDomManipulation.ts

```diff
@@ -1,85 +1,107 @@
 import { KartRun, LapSummary } from '../models/KartRun';
 
-export function addHeaderInfo(res: KartRun, changeCallBack: () => void) {
-  if (res?.lapSummaries?.[0] && res.driver) {
-    const name: HTMLElement | null = document.querySelector('#driver');
+export function addHeaderInfo(res: KartRun, resetCallBack: () => void) {
+  if (!(res?.driver && res.sessionName)) {
+    throw new Error('Error loading content');
+  }
 
-    if (!name) {
-      throw new Error('Seems like an error from our side!');
-    }
+  const name: HTMLElement | null = document.querySelector('#driver');
 
-    name.innerText = res.driver;
-    // Safety checks
-    const button = document.querySelector('button');
-    if (!button) {
-      throw new Error('Seems like an error from our side!');
-    }
+  if (!name) {
+    throw new Error('Seems like an error from our side!');
+  }
 
-    button.addEventListener('click', changeCallBack);
-
-    const select: HTMLElement | null = document.querySelector('select');
-    if (select) {
-      for (const lapIndex in res.lapSummaries) {
-        const option = document.createElement('option');
-        const valueAtr = document.createAttribute('value');
-        const lapNumber = Number(lapIndex) + 1;
-        valueAtr.value = `${lapNumber}`;
-        option.setAttributeNode(valueAtr);
-        option.innerText = `Lap ${lapNumber}`;
-        select.appendChild(option);
-      }
-    } else {
-      throw new Error('Seems like an error from our side!');
-    }
+  name.innerText = `${res.driver} - ${res.sessionName}`;
 
-    addLapInfo(res.lapSummaries[0]);
+  const resetButton = document.querySelector('#reset-btn');
+  if (!resetButton) {
+    throw new Error('Seems like an error from our side!');
```

Comment:
Maybe consider a more descriptive error message so that you can see where the error is. Having ` "Seems like an error from our side!" ` thrown from **many different** parts of the code can be a bit vague and hard to debug in the future